feat: move junit4 integration to org.testcontainers:junit4 module (#10285)
testcontainers/testcontainers-java: docs/quickstart/junit_4_quickstart.md

        <version>{{latest_version}}</version>
        <scope>test</scope>
    </dependency>
    <dependency>
        <groupId>org.testcontainers</groupId>
        <artifactId>junit4</artifactId>
        <version>{{latest_version}}</version>
        <scope>test</scope>
    </dependency>
    ```

## 2. Get Testcontainers to run a Redis container during our tests

Add the following to the body of our test class:

<!--codeinclude-->
[JUnit 4 Rule](../examples/junit4/redis/src/test/java/quickstart/RedisBackedCacheIntTest.java) inside_block:rule
<!--/codeinclude-->

The `@Rule` annotation tells JUnit to notify this field about various events in the test lifecycle.
Wrap the containers with `new TestContainersRule(...)` so the containers start and stop, according to the test lifecycle.

> In this case, our rule object is not `static`, so the container will start and stop with every test.

Reviewer: ```suggestion
Since the `redis` field is annotated with `@Rule` the container will start and stop with every test.
```
